crates/wasi-common/yanix/src/errno.rs

```diff
@@ -0,0 +1,270 @@
+//! Errno-specific for different Unix platforms
```

Comment:
Did you consider using `std::io::Error` instead of our own homebrew error type? Then you could also reuse `cvt` for syscall error handling.

Reply:
Actually, given that we are lower-level than potentially Rust itself, I'd prefer we've had our wrapper around the `libc` errors.

Reply:
And for that reason, I'm not able to reuse the `cvt` crate.